src/lightly_train/_commands/common_helpers.py

```diff
@@ -416,40 +444,67 @@ def get_dataset_mmap_filenames(
 
 
 def get_dataset(
-    data: PathLike | Dataset[DatasetItem],
+    data: PathLike | Sequence[PathLike] | Dataset[DatasetItem],
     transform: Transform,
     mmap_filepath: Path | None,
 ) -> Dataset[DatasetItem]:
     if isinstance(data, Dataset):
         logger.debug("Using provided dataset.")
         return data
 
-    data = Path(data).resolve()
-    logger.debug(f"Making sure data directory '{data}' exists and is not empty.")
-    if not data.exists():
-        raise ValueError(f"Data directory '{data}' does not exist!")
-    elif not data.is_dir():
-        raise ValueError(f"Data path '{data}' is not a directory!")
-    elif data.is_dir() and not any(data.iterdir()):
-        raise ValueError(f"Data directory '{data}' is empty!")
     if mmap_filepath is None:
         raise ValueError("Memory-mapped file path must be provided.")
 
-    logger.info(f"Initializing dataset from '{data}'.")
-    # NOTE(Guarin, 01/25): The bottleneck for dataset initialization is filename
-    # listing and not the memory mapping. Listing the train set from ImageNet takes
-    # about 30 seconds. This is mostly because os.walk is not parallelized.
-    filenames = image_dataset.list_image_filenames(image_dir=data)
     mask_dir = os.getenv(LIGHTLY_TRAIN_MASK_DIR)
-    return ImageDataset(
-        image_dir=data,
-        image_filenames=get_dataset_mmap_filenames(
-            filenames=filenames,
-            mmap_filepath=mmap_filepath,
-        ),
-        transform=transform,
-        mask_dir=Path(mask_dir) if mask_dir is not None else None,
-    )
+
+    if isinstance(data, (str, Path)):
+        data = Path(data).resolve()
+        if not data.exists():
+            raise ValueError(f"Data directory '{data}' does not exist!")
+        elif not data.is_dir():
+            raise ValueError(f"Data path '{data}' is not a directory!")
+        elif data.is_dir() and not any(data.iterdir()):
+            raise ValueError(f"Data directory '{data}' is empty!")
+        # Use relative paths as filenames when a single directory or file is provided to
+        # reduce the file size.
+        filenames = image_dataset.list_image_filenames(image_dir=data)
+        return ImageDataset(
+            image_dir=data,
+            image_filenames=get_dataset_mmap_filenames(
+                filenames=filenames,
+                mmap_filepath=mmap_filepath,
+            ),
+            transform=transform,
+            mask_dir=Path(mask_dir) if mask_dir is not None else None,
+        )
+
+    elif isinstance(data, Sequence) and not isinstance(data, str):
```

Comment:
We already check for `str` above

```suggestion
    elif isinstance(data, Sequence):
```